go.go

```diff
@@ -687,6 +781,54 @@ const goUnmarshalField = `{{if eq .Type "bool"}}
 		i++
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if header == {{.Index}} {
+	{{template "unmarshal-varint" .}}
+		if x > uint(ColferListMax) {
+			return 0, ColferMax(fmt.Sprintf("colfer: {{.String}} length %d exceeds %d elements", x, ColferListMax))
+		}
+
+		l := int(x)
+
+		if end := i + l*4; end >= len(data) {
+			i = end
+			goto eof
+		}
+		a := make([]int32, l)
+		for ai := range a {
+			if i+1 >= len(data) {
+				i++
+				goto eof
+			}
+
+			x := uint32(data[i])
+			i++
+	
+			if x >= 0x80 {
+				x &= 0x7f
+				for shift := uint(7); ; shift += 7 {
+					b := uint32(data[i])
+					i++
+					if i >= len(data) {
+						goto eof
+					}
+	
+					if b < 0x80 || shift == 56 {
```

Comment:
The `shift == 56` is only for 64-bit integers (to prevent the 10-th byte worst-case).

Reply:
fix it